packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,138 @@
+import { type Locator, type Page } from '@playwright/test';
+import { BasePage } from '@justeattakeaway/pie-webc-testing/src/helpers/page-object/base-page.ts';
+import {
+    cookieBanner,
+} from './selectors.ts';
+import {
+    PreferenceIds,
+} from '../../../src/defs.ts';
+
+const getPreferenceItemSelector = (id: PreferenceIds) => `#${id} [data-test-id="switch-component"]`;
+
+export class PieCookieBannerComponent extends BasePage {
+    readonly componentLocator: Locator;
+    readonly descriptionLocator: Locator;
+    readonly acceptAllButtonLocator: Locator;
+    readonly necessaryOnlyButtonLocator: Locator;
+    readonly managePrefsLinkLocator: Locator;
+    readonly bodyAcceptAllLinkLocator: Locator;
+    readonly bodyNecessaryOnlyLinkLocator;
+    readonly bodyManagePreferencesLinkLocator: Locator;
+    readonly bodyCookieStatementLinkLocator: Locator;
+    readonly bodyCookieTechnologiesLinkLocator: Locator;
+    readonly modalDescriptionLocator: Locator;
```

Comment:
I think you can also mark these as private if consumers don't need knowledge of them :)